src/tools/r2rdump/R2RDump.cs

```diff
@@ -32,6 +34,44 @@ private R2RDump()
         {
         }
 
+        private void PrintToWriter(IntPtr pMsg, long arg0, long arg1, long arg2, long arg3, long arg4, long arg5, long arg6, long arg7)
+        {
+            string fmt = Marshal.PtrToStringAnsi(pMsg);
+            long[] args = { arg0, arg1, arg2, arg3, arg4, arg5, arg6, arg7 };
+            _writer.WriteLine(Sprintf(fmt, args));
+        }
+
+        private void PrintToConsole(IntPtr pMsg, long arg0, long arg1, long arg2, long arg3, long arg4, long arg5, long arg6, long arg7)
+        {
+            string fmt = Marshal.PtrToStringAnsi(pMsg);
+            long[] args = { arg0, arg1, arg2, arg3, arg4, arg5, arg6, arg7 };
+            Console.WriteLine(Sprintf(fmt, args));
+        }
+
+        private string Sprintf(string fmt, long[] args)
```

Comment:
As we discussed in your office, it would be great if we can do the sprintf on the C++ side and just pass the formatted string over to managed. Then we don't need to worry about making our own matching implementation.